src/coreclr/jit/importer.cpp

```diff
@@ -14021,6 +14021,137 @@ void Compiler::impImportBlockCode(BasicBlock* block)
                         // Try and fold the introduced cast
                         op1 = gtFoldExprConst(op1);
                     }
+
+#if defined(TARGET_XARCH)
+                    if (!ovfl && op1->OperIs(GT_CAST) && varTypeIsFloating(op1->AsCast()->CastOp()) && varTypeIsIntegral(lclTyp))
```

Comment:
That's a "more complex" change and this is far from the opcode importation logic.

While I do think it would be good to break apart `impImportBlockCode` (seeing as its a nearly 6000 line method), I don't think this PR is the place to do it.